l1-contracts/deploy-scripts/DeployGatewayTransactionFilterer.s.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {Script, console2 as console} from "forge-std/Script.sol";
+import {Create2FactoryUtils} from "./Create2FactoryUtils.s.sol";
+import {GatewayTransactionFilterer} from "contracts/transactionFilterer/GatewayTransactionFilterer.sol";
+import {TransparentUpgradeableProxy} from "@openzeppelin/contracts-v4/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {IBridgehub} from "contracts/bridgehub/IBridgehub.sol";
+import {stdToml} from "forge-std/StdToml.sol";
+
+/// @title DeployGatewayTransactionFilterer
+/// @notice This script deploys a GatewayTransactionFilterer behind a TransparentUpgradeableProxy,
+/// using Create2 with notify. It takes five parameters:
+/// - bridgehub: The address of the Bridgehub contract.
+/// - chainAdmin: The admin to be set as the initial owner of the deployed filterer.
+/// - chainProxyAdmin: The pre-deployed proxy admin that manages the proxy.
+/// - create2FactoryAddress: The configured Create2 factory address.
+/// - create2FactorySalt: The salt for the Create2 deployment.
+contract DeployGatewayTransactionFilterer is Script, Create2FactoryUtils {
+    using stdToml for string;
+
+    function run(
+        address bridgehub,
+        address chainAdmin,
+        address chainProxyAdmin,
+        address create2FactoryAddress,
+        bytes32 create2FactorySalt
+    ) public returns (address proxy) {
+        // Initialize and instantiate Create2Factory before any deployment.
+        _initCreate2FactoryParams(create2FactoryAddress, create2FactorySalt);
+        instantiateCreate2Factory();
+
+        // Query the L1 asset router from the Bridgehub.
+        address l1AssetRouter = IBridgehub(bridgehub).assetRouter();
+
+        // Deploy the GatewayTransactionFilterer implementation via Create2 with notify.
+        // The constructor of GatewayTransactionFilterer expects (bridgehub, l1AssetRouter).
+        address implementation = deployViaCreate2AndNotify(
+            type(GatewayTransactionFilterer).creationCode,
+            abi.encode(bridgehub, l1AssetRouter),
+            "GatewayTransactionFilterer",
+            "GatewayTransactionFilterer"
+        );
+
+        // Prepare the initialization calldata.
+        // This initialize function sets the transaction filterer's initial owner to chainAdmin.
+        bytes memory initData = abi.encodeCall(GatewayTransactionFilterer.initialize, (chainAdmin));
+
+        // Deploy the TransparentUpgradeableProxy via Create2 with notify.
+        // The proxy uses the provided chainProxyAdmin as its admin.
+        proxy = deployViaCreate2AndNotify(
```

Comment:
It might make sense to deploy this contract using the Create2FactoryUtils patters, and set the initData is a standard location with the other initData